src/SegmentedControl/SegmentedControl.tsx

```diff
@@ -14,20 +14,20 @@ type SegmentedControlProps = {
   'aria-labelledby'?: string
   'aria-describedby'?: string
   /** Whether the control fills the width of its parent */
-  fullWidth?: boolean
+  fullWidth?: boolean | Partial<Record<WidthOnlyViewportRangeKeys, boolean>>
```

Comment:
You could use the `ResponsiveValue<>` utility here:

```suggestion
  fullWidth?: boolean | ResponsiveValue<boolean>
```